unittests/FEXLinuxTests/tests/signal/Syscall_state.64.cpp

```diff
@@ -0,0 +1,174 @@
+#include <catch2/catch.hpp>
+#include <csetjmp>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <signal.h>
+#include <string.h>
+#include <sys/wait.h>
+
+struct CPUState {
+  uint64_t Registers[16];
+
+  uint64_t eflags;
+};
+
+CPUState CapturedState{};
+extern "C" uint64_t HLT_INST;
```

Comment:
Could probably do with a comment explaining that this refers to the label defined in the inline assembly below.